docs/ebpf/concepts/elf-sections.md

```diff
@@ -0,0 +1,207 @@
+    definitions](../maps/declaring.md).
+
+`maps`
+
+:   **Deprecated.** This section is expected to only contain fixed-width `struct
+    bpf_map_def` variables. Larger structs like iproute2's `struct bpf_elf_map`
+    can also be used for backwards compatibility. Any extra bytes past the end
+    of the size of a `struct bpf_map_def` are exposed by {{
+    godoc('MapSpec.Extra') }} and must be drained before attempting to create
+    the Map.
+
+#### :material-head-cog: Advanced: Special Map Sections
+
+`.data`
+
+:   The LLVM BPF backend implements accesses to mutable global variables as
+    direct Array Map accesses. Since a single BPF program can be executed
+    concurrently as a result of the kernel processing packets and other events
+    asynchronously, `.data` and the global variables it represents are
+    considered shared memory.
+
+    This section is exposed by {{ godoc('CollectionSpec.Maps') }} as a
+    single-element BPF array and its contents are accessible through {{
+    godoc('MapSpec.Contents') }}. Its layout is described by its {{
+    godoc('MapSpec.Value') }}, a {{ godoc('btf/Datasec') }} containing all
+    global variables in the compilation unit.
+
+    The contents of the Map may be modified to control the default values
+    used for the eBPF program's global variables.
+
+`.rodata*`
+
+:   Like `.data`, but for constants. This is a prefix and matches sections like
+    `.rodata.foo`. Constants can be emitted to different sections using e.g.
+    `#!c SEC(".rodata.foo") volatile const foobar = 123;`. This can prove useful
+    for isolating certain constants to well-known sections for Go code
+    generation or custom constant rewriting logic.
+
+`.bss`
+
+:   Section emitted by the compiler when zero-initialized globals are present in
+    the ELF. Is typically zero-length. Exposed by {{
+    godoc('CollectionSpec.Maps') }}, but not really useful.
+
+`.rel*`
+
+:   Not exposed by {{ proj }}, only used behind the scenes. Relocation sections
+    contain relocation records against their non-`.rel` prefixed counterparts.
+    This is mainly used for fixing up BPF instructions referring to Maps and
+    global variables.
+
+### Program Sections
+
+Names of Program sections mainly define the program's {{ godoc('ProgramType')
+}}, but also its {{ godoc('AttachType') }} and {{ godoc('AttachFlags') }} are
+automatically set for convenience based on its section name.
+
+As described previously, section prefixes containing a forward slash `/` expect
+a second component to follow the slash. For example, a program in the
+`kprobe/slub_flush` section will automatically have its {{
+godoc('ProgramSpec.AttachTo') }} field set to `slub_flush` to facilitate
+attaching the program later on.
+
+Additionally, the program's original full section name can be found in {{
+godoc('ProgramSpec.SectionName') }}.
+
+| Section (Prefix)      | {{ godoc('ProgramType') }} | {{ godoc('AttachType') }}        | {{ godoc('AttachFlags') }} |
```

Comment:
There's a note in the code referring to this page. I always need to grep through the code to find this table. It's somewhere in the middle of an unexported function near the bottom of elf_reader.go, so not really practical to generate a permalink to. If this would be in its own file, maybe.

Everything we put in docs itself is indexed by the mkdocs search engine and will show up on Google as well. I think this provides the most value. We add section prefixes 1 or 2 times a year, we're good.

Reply:
Then we should just link to https://docs.kernel.org/bpf/libbpf/program_types.html

Reply:
Oh, didn't know about that page, I'll add a link to it. Still, 2 reasons why we should keep a table here:
- if we link to libbpf documentation, the constants they use are different, which may still confuse the user
- we usually lag behind implementing new prog types / section names, so the user will still be surprised if they see a section name in the libbpf docs that doesn't work in ebpf-go

If we maintain a table, it's just clear which sections we support and no further digging is required.